diff --git a/dist/IO/DWG/DwgStreamReaders/DwgObjectReader.js b/dist/IO/DWG/DwgStreamReaders/DwgObjectReader.js
index 76143736c81f492d09c27def63a404575852ad2a..584a409e80755565792529c1982b155e066d2ef8 100644
--- a/dist/IO/DWG/DwgStreamReaders/DwgObjectReader.js	
+++ b/dist/IO/DWG/DwgStreamReaders/DwgObjectReader.js
@@ -247,7 +247,7 @@ export class DwgObjectReader extends DwgSectionIO {
             this._classes.set(c.classNumber, c);
         }
         this._memoryStream = new Uint8Array(reader.stream);
-        this._crcReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream));
+        this._crcReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream);
     }
     read() {
         while (this._handles.length > 0) {
@@ -291,20 +291,20 @@ export class DwgObjectReader extends DwgSectionIO {
         if (this.r2010Plus) {
             const handleSize = this._crcReader.readModularChar();
             const handleSectionOffset = this._crcReader.positionInBits() + sizeInBits - handleSize;
-            this._objectReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._objectReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._objectReader.setPositionInBits(this._crcReader.positionInBits());
             this._objectInitialPos = this._objectReader.positionInBits();
             type = this._objectReader.readObjectType();
-            this._handlesReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._handlesReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._handlesReader.setPositionInBits(handleSectionOffset);
-            this._textReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._textReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._textReader.setPositionByFlag(handleSectionOffset - 1);
             this._mergedReaders = new DwgMergedReader(this._objectReader, this._textReader, this._handlesReader);
         }
         else {
-            this._objectReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._objectReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._objectReader.setPositionInBits(this._crcReader.positionInBits());
-            this._handlesReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._handlesReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._textReader = this._objectReader;
             this._objectInitialPos = this._objectReader.positionInBits();
             type = this._objectReader.readObjectType();
@@ -532,7 +532,7 @@ export class DwgObjectReader extends DwgSectionIO {
         const size = this._objectReader.readRawLong();
         this._handlesReader.setPositionInBits(size + this._objectInitialPos);
         if (this._version === ACadVersion.AC1021) {
-            this._textReader = DwgStreamReaderBase.getStreamHandler(this._version, new Uint8Array(this._memoryStream), this._reader.encoding);
+            this._textReader = DwgStreamReaderBase.getStreamHandler(this._version, this._memoryStream, this._reader.encoding);
             this._textReader.setPositionByFlag(size + this._objectInitialPos - 1);
         }
         this._mergedReaders = new DwgMergedReader(this._objectReader, this._textReader, this._handlesReader);
